--- setup.py.orig	2022-12-06 19:31:21.000000000 +0100
+++ setup.py	2022-12-16 13:14:39.256469721 +0100
@@ -2272,7 +2272,8 @@
                                libraries=['m']))
 
         ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
-        ffi_lib = None
+        ffi_lib = ":libffi_pic.a"
+        extra_link_args.append('--exclude-libs,libffi_pic.a')
 
         ffi_inc_dirs = self.inc_dirs.copy()
         if MACOS:
